api/providers/SupplierProvider.cs

```diff
@@ -15,6 +15,14 @@ public SupplierProvider(AppDbContext db, IValidator<Supplier> validator, Contact
         _contactProvider = contactProvider;
     }
 
+    public List<Item> GetItemsBySupplierId(Guid id)
+    {
+        var itemsSpecificSupplier = _db.Items.Where(l => l.SupplierId == id).ToList();
+        if (itemsSpecificSupplier.Count == 0)
+            throw new ApiFlowException("No items found for this supplier");
+
+        return itemsSpecificSupplier;
+    }
```

Comment:
Dit kan verkort worden naar een arrow function en die checks zijn niet nodig

```suggestion
public List<Item>? GetItemsBySupplierId(Guid id) => _db.Items.Where(l => l.SupplierId == id).ToList();
```